### PR TITLE
Add the ability to only advertise local routes or don't advertise at all

### PR DESCRIPTION
**Pull request by asim**
This PR adds two new router advertise strategies; Local and None.

- AdvertiseLocal will only advertise the local routes. Related to https://github.com/micro/go-micro/issues/927
- AdvertiseNone will avoid advertising routes completely.

We now need to add the ability to set this as a flag/env var

### CODE REVIEW

File: router/default.go [hunk 1]
```diff
@@ -728,32 +750,42 @@ func (r *router) flushRouteEvents(evType EventType) ([]*Event, error) {
...
+		if current.Metric == route.Metric {
 			if route.Router == r.options.Id {
 				bestRoutes[routeKey] = route
 				continue
 			}
 		}
 	}
 
-	log.Debugf("Router advertising %d best routes out of %d", len(bestRoutes), len(routes))
+	log.Debugf("Router advertising %d %s routes out of %d", len(bestRoutes), r.options.Advertise.String(), len(routes))
```
Comment (milosgajdos) [inline router/default.go]: `.String()` is not required here as `%s` will explicitly call `.String()`

File: router/default.go [hunk 2]
```diff
@@ -426,6 +430,12 @@ func (r *router) advertiseEvents() error {
 					continue
 				}
 
+				// if we only advertise local routes delete and bail out
+				if local && advert.event.Route.Link != "local" {
```
Comment (milosgajdos) [inline router/default.go]: Makes me wonder whether we should skip adding the routes to the adverts map in the first place as soon as they're received.

File: router/default.go [hunk 3]
```diff
@@ -714,6 +730,12 @@ func (r *router) flushRouteEvents(evType EventType) ([]*Event, error) {
 		return nil, fmt.Errorf("failed listing routes: %s", err)
 	}
 
+	// Do not advertise anything
+	if r.options.Advertise == AdvertiseNone {
```
Comment (milosgajdos) [inline router/default.go]: Shouldnt this be called right the first thing i.e. before we even call `List()` on router?

File: router/default.go [hunk 4]
```diff
@@ -436,6 +446,12 @@ func (r *router) advertiseEvents() error {
 				delete(adverts, key)
 			}
 
+			// If we're not advertising any events then throw them away.
+			// We do this here because someone may change the strategy at any time.
+			if r.options.Advertise == AdvertiseNone {
```
Comment (milosgajdos) [inline router/default.go]: Shouldnt we call this before we start iterating over the adverts?

File: router/default.go (hunk 3 above)
Comment (asim) [reply router/default.go]: Good spot

File: router/default.go (hunk 4 above)
Comment (asim) [reply router/default.go]: Done

File: router/default.go (hunk 2 above)
Comment (asim) [reply router/default.go]: Yea good idea, done.

**Pull request merged**